src/db_adapters/receipts.rs

```diff
@@ -129,6 +66,105 @@ pub(crate) async fn store_receipts(
+                .inner_join(
+                    schema::receipts::table.on(
+                        schema::execution_outcome_receipts::dsl::execution_outcome_receipt_id
+                            .eq(schema::receipts::dsl::receipt_id),
+                    ),
+                )
+                .filter(
+                    schema::execution_outcome_receipts::dsl::receipt_id.eq(any(receipts.clone())),
+                )
+                .select((
+                    schema::execution_outcome_receipts::dsl::receipt_id,
+                    schema::receipts::dsl::transaction_hash,
+                ))
+                .load_async(&pool)
+                .await
+            {
+                Ok(res) => {
+                    break res;
+                }
+                Err(async_error) => {
+                    error!(
+                        target: crate::INDEXER_FOR_EXPLORER,
+                        "Error occurred while fetching the parent receipt for Receipt. Retrying in {} milliseconds... \n {:#?}",
+                        crate::INTERVAL.as_millis(),
+                        async_error,
+                    );
+                    tokio::time::delay_for(crate::INTERVAL).await;
+                }
+            }
+        };
+
+        tx_hashes_for_receipts.extend(tx_hashes_for_receipts_via_outcomes);
+
+        if tx_hashes_for_receipts.len() == receipt_ids.len() {
+            break;
+        }
+
+        receipts.retain(|r| tx_hashes_for_receipts.contains_key(r.as_str()));
+
+        let tx_hashes_for_receipt_via_transactions: Vec<(String, String)> = loop {
+            match schema::transactions::table
+                .filter(schema::transactions::dsl::receipt_id.eq(any(receipts.clone())))
+                .select((
+                    schema::transactions::dsl::receipt_id,
+                    schema::transactions::dsl::transaction_hash,
+                ))
+                .load_async(&pool)
+                .await
+            {
+                Ok(res) => {
+                    break res;
+                }
+                Err(async_error) => {
+                    error!(
+                        target: crate::INDEXER_FOR_EXPLORER,
+                        "Error occurred while fetching the parent transaction for ExecutionOutcome. Retrying in {} milliseconds... \n {:#?}",
+                        crate::INTERVAL.as_millis(),
+                        async_error,
+                    );
+                    tokio::time::delay_for(crate::INTERVAL).await;
+                }
+            }
+        };
+
+        tx_hashes_for_receipts.extend(tx_hashes_for_receipt_via_transactions);
+
+        if tx_hashes_for_receipts.len() == receipt_ids.len() {
+            break;
+        }
+
+        receipts.retain(|r| tx_hashes_for_receipts.contains_key(r.as_str()));
+
+        if !strict_mode {
+            if retries_left > 0 {
+                retries_left = retries_left.saturating_sub(1);
```

Comment:
```suggestion
                retries_left -= 1;
```

There is no need for `saturating_sub` since you explicitly check for `> 0`